Sources/XcodeGenKit/CarthageVersionLoader.swift

```diff
@@ -9,24 +9,52 @@ import Foundation
 import PathKit
 import ProjectSpec
 
+class Mutex<T> {
+    var value: T
+    //var mutex: pthread_mutex_t = pthread_mutex_t()
+    var semaphore: DispatchSemaphore = DispatchSemaphore(value: 1)
+    init(_ value: T) {
+        self.value = value
+        //pthread_mutex_init(&mutex, nil)
+    }
+
+    func get<U>(closure: (inout T) throws -> (U)) rethrows -> U {
+        semaphore.wait()
+        defer { semaphore.signal() }
+        //pthread_mutex_lock(&mutex)
+        //defer { pthread_mutex_unlock(&mutex) }
+        let newValue = try closure(&value)
+        //value = newValue
+        return newValue
+    }
+
+    func get(closure: (inout T) -> ()) {
+        semaphore.wait()
+        closure(&value)
+        semaphore.signal()
+    }
+}
+
 class CarthageVersionLoader {
 
     private let buildPath: Path
-    private var cachedFiles: [String: CarthageVersionFile] = [:]
+    private var cachedFilesMutex: Mutex<[String: CarthageVersionFile]> = Mutex([:])
```

Comment:
note that we'll need to be careful in the future not to add any non-thread-safe state to CarthageVersionLoader. for now it's just `cachedFiles` that needs this, but something to keep in mind

Reply:
Let's add a comment to `CarthageVersionLoader` to warn future us.